src/runtime_src/core/common/info_platform.cpp

```diff
@@ -0,0 +1,237 @@
+  try {
+    mac_contiguous_num = xrt_core::device_query<xrt_core::query::mac_contiguous_num>(dev);
+    mac_addr_first = xrt_core::device_query<xrt_core::query::mac_addr_first>(dev);
+  }
+  catch (const xrt_core::query::no_such_key&) {
+    // Ignoring if not available: Edge Case 
+  }
+
+  //new flow
+  if (mac_contiguous_num && !mac_addr_first.empty()) {
+    std::string mac_prefix = mac_addr_first.substr(0, mac_addr_first.find_last_of(":"));
+    std::string mac_base = mac_addr_first.substr(mac_addr_first.find_last_of(":") + 1);
+    std::stringstream ss;
+    uint32_t mac_base_val = 0;
+    ss << std::hex << mac_base;
+    ss >> mac_base_val;
+
+    for (uint32_t i = 0; i < (uint32_t)mac_contiguous_num; i++) {
+      boost::property_tree::ptree addr;
+      auto base = boost::format("%02X") % (mac_base_val + i);
+      addr.add("address", mac_prefix + ":" + base.str());
+      ptree.push_back(std::make_pair("", addr));
+    } 
+  }
+  else { //old flow
+    std::vector<std::string> mac_addr;
+    try {	  
+      mac_addr = xrt_core::device_query<xrt_core::query::mac_addr_list>(dev);
+    }
+    catch (const xrt_core::query::no_such_key&) {
+      // Ignoring if not available: Edge Case 
+    }
+    for (const auto& a : mac_addr) {
+      boost::property_tree::ptree addr;
+      if (!a.empty() && a.compare("FF:FF:FF:FF:FF:FF") != 0) {
+        addr.add("address", a);
+        ptree.push_back(std::make_pair("", addr));
+      }
+    }
+  }
+
+  return ptree;
+}
+
+} //unnamed namespace
+
+namespace xrt_core {
+namespace platform {
+
+boost::property_tree::ptree
+platform_info(const xrt_core::device * device) {
+  boost::property_tree::ptree ptree;
+  boost::property_tree::ptree pt_platform;
+  
+  boost::property_tree::ptree static_region;
+  static_region.add("vbnv", xrt_core::device_query<xrt_core::query::rom_vbnv>(device));
+
+  std::vector<std::string> interface_uuids;
+  // the vectors are being populated by empty strings which need to be removed
+  try {
+    interface_uuids = xrt_core::device_query<xrt_core::query::interface_uuids>(device);
+    interface_uuids.erase(
+      std::remove_if(interface_uuids.begin(), interface_uuids.end(),	
+                  [](const std::string& s) { return s.empty(); }), interface_uuids.end());
+  } catch (const xrt_core::query::no_such_key&) {}
+  static_region.add("interface_uuid", xrt_core::utils::string_to_UUID(interface_uuids[0]));
+
+  try {
+    static_region.add("jtag_idcode", xrt_core::query::idcode::to_string(xrt_core::device_query<xrt_core::query::idcode>(device)));
+  }
+  catch (const xrt_core::query::no_such_key&) {
+    // Ignoring if not available: Edge Case
+    static_region.add("jtag_idcode", "N/A");
+  }
+
+  try {
+    static_region.add("fpga_name", xrt_core::device_query<xrt_core::query::rom_fpga_name>(device));
+  }
+  catch (const xrt_core::query::no_such_key&) {
+    // Ignoring if not available: Edge Case 
+    static_region.add("fpga_name", "N/A");
+  }
```

Comment:
This is the same pattern used.  I think this could be namespace { .. helper function .. }, that would be one way of a little bit of refactoring.

Same helper function can be used below also, I get very dizzy looking at all these try/catch instances.